patches/server/0616-Enhance-console-tab-completions-for-brigadier-comman.patch

```diff
@@ -48,28 +53,25 @@ index 0000000000000000000000000000000000000000..5fee12b106a50f7c92c9ecf8256d5e12
 +import org.jline.reader.LineReader;
 +import org.jline.reader.ParsedLine;
 +
-+import java.util.ArrayList;
-+import java.util.Collections;
-+import java.util.List;
-+
 +import static com.destroystokyo.paper.event.server.AsyncTabCompleteEvent.Completion.completion;
 +
 +public final class BrigadierCommandCompleter {
-+    private final CommandSourceStack commandSourceStack;
++    private final Supplier<CommandSourceStack> commandSourceStack;
 +    private final DedicatedServer server;
 +
-+    public BrigadierCommandCompleter(final @NonNull DedicatedServer server, final @NonNull CommandSourceStack commandSourceStack) {
++    public BrigadierCommandCompleter(final @NonNull DedicatedServer server) {
 +        this.server = server;
-+        this.commandSourceStack = commandSourceStack;
++        this.commandSourceStack = Suppliers.memoize(this.server::createCommandSourceStack);
 +    }
 +
 +    public void complete(final @NonNull LineReader reader, final @NonNull ParsedLine line, final @NonNull List<Candidate> candidates, final @NonNull List<Completion> existing) {
-+        if (!io.papermc.paper.configuration.GlobalConfiguration.get().console.enableBrigadierCompletions) {
++        //noinspection ConstantConditions
++        if (this.server.overworld() == null || !io.papermc.paper.configuration.GlobalConfiguration.get().console.enableBrigadierCompletions) { // check if overworld is null, as worlds haven't been loaded yet
```

Comment:
I think the overworld check should go first and not call addCandidates, to avoid inconsistency between bukkit and brigadier commands